Functionality_Tests/FunctionalityTests.cs

```diff
@@ -10,32 +10,54 @@
 using SeleniumExtras.WaitHelpers;
 using OpenQA.Selenium.DevTools.V115.Page;
 using Functionality_Tests_Suit.FactoryPattern;
+using OpenQA.Selenium.DevTools.V115.FedCm;
 
 namespace Functionality_Tests_Suit
 {
     [FixtureLifeCycle(LifeCycle.InstancePerTestCase)]
     [Parallelizable(scope: ParallelScope.Self)]
     public class FunctionalityTests : BaseTest
     {
-        [Test, Order(1)]
+        private bool needToRemoveFromTheCart;
+        private List<string> itemsToRemove = new List<string>();
+
+        [SetUp]
+        public void SetUp()
+        {
+            StandardUserLogin();
+        }
+
+        [Test]
         public void PriceSorting()
         {
-            SuccessfulLogin();
-            var elementItemSorting = Driver.FindElement(By.TagName("select"));
-            elementItemSorting.Click();
-            var elementPriceSorting = Driver.FindElement(By.CssSelector("[value='lohi']"));
-            elementPriceSorting.Click();
+            var selectElement = Driver.FindElement(By.TagName("select"));
+            selectElement.Click();
+            var select = new SelectElement(selectElement);
+            select.SelectByValue("lohi");
+
             var elementMinPrice = Driver.FindElement(By.CssSelector("[class='inventory_list']:first-child"));
             Assert.That(elementMinPrice.Text.Contains("7.99"));
 
             var elementItemsList = Driver.FindElements(By.ClassName("inventory_list"));
             Assert.That(elementItemsList, Is.Ordered.Ascending);
         }
 
-        [Test, Order(2)]
-        public void RemoveFromCart()
+        [Test]
+        public void AddToCart()
+        {
+            AddProductToCart();
+            needToRemoveFromTheCart = true;
+            itemsToRemove.Add("1");
+        }
+        [Test]
+        public void AddAndRemoveFromCart()
         {
             AddProductToCart();
+            RemoveItemFromCart();
+        }
+
+        public void RemoveItemFromCart()
+        {
             var elementShopingCartIcon = Driver.FindElement(By.ClassName("shopping_cart_link"));
```

Comment:
When tests run on remote machines they have some attempts
For example,

Attempt of run 1: 
Test AddToCart failed, and AddAndRemoveFromCart test also failed, but failed only after adding product to the cart, but before deleting it from the cart.

Attempt of run 2: 
When test AddToCart  will be running, our product will be in the cart, because it was added in RemoveItemFromCart test and was not deleted from it.

So, please, make needToRemoveFromTheCart = true from the cart in AddAndRemoveFromCart test right after adding product to the cart. If it will be deleted in test - okay, if not - we will delete in in TearDown, but also we need to add check if something is in the card it not. (if deleteButton.Displayed  => click deleteButton)

Reply:
done